### PR TITLE
fix(next-server): 'quiet' setting delegate for custom server

### PR DESCRIPTION
**Pull request by bobaaaaa**
## Info

I did not create a GitHub issue, but this PR for fixing the issue. Hope that's okay.

## The bug

We run a 'normal' custom server described here: https://nextjs.org/docs/pages/building-your-application/configuring/custom-server

```ts
const next = require('next');
const dev = process.env.NODE_ENV !== 'production';
const port = 3000;
const app = next({ dev, port, quiet: !dev })
const handle = app.getRequestHandler()
// ...
```

The `quiet` settings is missed in some deeper Next.js internal api calls. For that reason, the setting did not work at all. This works only in production mode.

## Testing

I tried to implement a test in `/test/production/custom-server` but I failed to get the `stderr` messages. When someone give me some guidance, I will add a new test.

Here are some criteria:

- need: custom server with `quiet: true` setting
- need: a next.js page with thrown error. e.g. `export const getServerSideProps = () => { throw new Error('failed') }`
- need: collect all `stderr` messages

I tested the fix in our project by editing the files in `node_modules` -- it worked.

## Background

We upgrade our project from next `13.3.4` to `14.2.0` and saw this regression. I can not tell in which version this bug was added. We run a Next.js + 'custom server' + Docker setup. Because we have a lot of traffic, we need to suppress the standard Next.js logging and using our own logging.

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 239c5ec6a523e3eab8386245448d5019efb6d71c

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Review by ijjk — COMMENTED** [review]
Hi, could we add a test for this similar to our custom server test?

x-ref: https://github.com/vercel/next.js/blob/canary/test/production/custom-server/custom-server.test.ts

**Comment by bobaaaaa** [comment]
@ijjk I wrote this in the PR description. I tried this in exact this spec, but I failed to get the `stderr` message collecting to work. When someone give me a guidance, I will do it

**Comment by ijjk** [comment]
Could you push the test you attempted to add and we can work from there?

File: test/production/custom-server/custom-server.test.ts [hunk 1]
```diff
@@ -15,6 +15,10 @@ createNextDescribe(
       expect($('p').text()).toBe(`Page ${page}`)
     })
 
+    it('should not log any error messages when server is started with "quiet" setting', async () => {
+      await next.render(`/error`)
+    })
```
Comment (bobaaaaa) [inline test/production/custom-server/custom-server.test.ts:20]: @ijjk My first impl. was here: https://github.com/vercel/next.js/blob/canary/test/integration/custom-server/test/index.test.js That was super simple. But not for production mode :(

File: test/production/custom-server/server.js [hunk 2]
```diff
@@ -2,12 +2,13 @@ const { createServer } = require('http')
 const { parse } = require('url')
 const next = require('next')
 const getPort = require('get-port')
+const quiet = process.env.USE_QUIET === 'true'
```
Comment (bobaaaaa) [inline test/production/custom-server/server.js:5]: from here: https://github.com/vercel/next.js/blob/64da71cfc982e485d53d93c25305080115cc6d24/test/integration/custom-server/server.js#L15

File: test/production/custom-server/custom-server.test.ts (hunk 1 above)
Comment (ijjk) [reply test/production/custom-server/custom-server.test.ts:20]: They can be set via the `env` field next to `startCommand` then the CLI output can be accessed via `next.cliOutput`.

Comment (bobaaaaa) [reply test/production/custom-server/custom-server.test.ts]: @ijjk Much appreciated for the help. I added two test cases:

- `quite: true` -> no logging
- default `quite` -> logging

**Review by ijjk — APPROVED** [review]

**Pull request merged**